### ISSUE TITLE
[package] ffmpeg/4.4.4: Build failed

### ISSUE THREAD

**Issue by nagayev** [conan-io/conan-center-index#24034]
### Description

1. Try to install ffmpeg/4.4.4 with disabled ssl.
2. Get `Error: operand type mismatch for 'shr' `

See https://github.com/ffvvc/FFmpeg/issues/116

### Package and Environment Details

* Package Name/Version: **ffmpeg/4.4.4**
* Operating System+version: **Linux Ubuntu 24.04**
* Compiler+version: **GCC 13.2**
* Conan version: **conan 2.3.1**
* Python version: **Python 3.7.4**

### Conan profile

arch=x86_64
build_type=Release
compiler=gcc
compiler.cppstd=gnu[17](https://github.com/pslns/positiverep/actions/runs/9151206575/job/25156920787#step:6:18)
compiler.libcxx=libstdc++11
compiler.version=13
os=Linux

### Steps to reproduce

conan install . -c tools.system.package_manager:mode=install -c tools.system.package_manager:sudo=True --build=missing -of build

### Logs

<details><summary>Click to expand log</summary>

```
2024-05-20T00:10:36.1049629Z ffmpeg/4.4.4: RUN: make -j2
2024-05-20T00:10:36.2480675Z GEN	libavutil/libavutil.version
2024-05-20T00:10:36.2621004Z GEN	libavcodec/libavcodec.version
2024-05-20T00:10:36.2739643Z GEN	libavformat/libavformat.version
2024-05-20T00:10:36.2797718Z GEN	libavfilter/libavfilter.version
2024-05-20T00:10:36.2895990Z GEN	libavdevice/libavdevice.version
2024-05-20T00:10:36.2947609Z GEN	libswscale/libswscale.version
2024-05-20T00:10:36.2957569Z GEN	libswresample/libswresample.version
2024-05-20T00:10:36.3019247Z GEN	libpostproc/libpostproc.version
2024-05-20T00:10:36.4320287Z CC	libavdevice/alldevices.o
2024-05-20T00:10:36.4326012Z CC	libavdevice/alsa.o
2024-05-20T00:10:36.6281049Z CC	libavdevice/alsa_dec.o
2024-05-20T00:10:36.7433684Z CC	libavdevice/alsa_enc.o
2024-05-20T00:10:36.7939075Z CC	libavdevice/avdevice.o
2024-05-20T00:10:36.9110342Z CC	libavdevice/fbdev_common.o
2024-05-20T00:10:36.9388267Z CC	libavdevice/fbdev_dec.o
2024-05-20T00:10:37.0607561Z CC	libavdevice/fbdev_enc.o
2024-05-20T00:10:37.1219141Z CC	libavdevice/lavfi.o
2024-05-20T00:10:37.2338384Z CC	libavdevice/oss.o
2024-05-20T00:10:37.3805244Z CC	libavdevice/oss_dec.o
2024-05-20T00:10:37.4024903Z CC	libavdevice/oss_enc.o
2024-05-20T00:10:37.5287327Z CC	libavdevice/pulse_audio_common.o
2024-05-20T00:10:37.5394435Z CC	libavdevice/pulse_audio_dec.o
2024-05-20T00:10:37.7115542Z CC	libavdevice/pulse_audio_enc.o
2024-05-20T00:10:37.7495087Z CC	libavdevice/timefilter.o
2024-05-20T00:10:37.8377939Z CC	libavdevice/utils.o
2024-05-20T00:10:37.9478959Z CC	libavdevice/v4l2-common.o
2024-05-20T00:10:38.0961093Z CC	libavdevice/v4l2.o
2024-05-20T00:10:38.1147175Z CC	libavdevice/v4l2enc.o
2024-05-20T00:10:38.2781837Z CC	libavdevice/xcbgrab.o
2024-05-20T00:10:38.3121189Z src/libavdevice/v4l2.c: In function ‘v4l2_get_device_list’:
2024-05-20T00:10:38.3127311Z src/libavdevice/v4l2.c:1054:58: warning: ‘%s’ directive output may be truncated writing up to 255 bytes into a region of size 251 [-Wformat-truncation=]
2024-05-20T00:10:38.3129745Z  1054 |         snprintf(device_name, sizeof(device_name), "/dev/%s", entry->d_name);
2024-05-20T00:10:38.3131062Z       |                                                          ^~
2024-05-20T00:10:38.3132461Z In file included from /usr/include/stdio.h:980,
2024-05-20T00:10:38.3133482Z                  from src/libavformat/avformat.h:311,
2024-05-20T00:10:38.3134478Z                  from src/libavformat/internal.h:27,
2024-05-20T00:10:38.3135588Z                  from src/libavdevice/v4l2-common.h:24,
2024-05-20T00:10:38.3136662Z                  from src/libavdevice/v4l2.c:35:
2024-05-20T00:10:38.3137598Z In function ‘snprintf’,
2024-05-20T00:10:38.3138421Z     inlined from ‘v4l2_get_device_list’ at src/libavdevice/v4l2.c:1054:9:
2024-05-20T00:10:38.3140187Z /usr/include/x86_64-linux-gnu/bits/stdio2.h:54:10: note: ‘__builtin___snprintf_chk’ output between 6 and 261 bytes into a destination of size 256
2024-05-20T00:10:38.3141994Z    54 |   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
2024-05-20T00:10:38.3142892Z       |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2024-05-20T00:10:38.3143644Z    55 |                                    __glibc_objsize (__s), __fmt,
2024-05-20T00:10:38.3144377Z       |                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2024-05-20T00:10:38.3145138Z    56 |                                    __va_arg_pack ());
2024-05-20T00:10:38.3145822Z       |                                    ~~~~~~~~~~~~~~~~~
2024-05-20T00:10:38.6073345Z CC	libavdevice/xv.o
2024-05-20T00:10:38.6696464Z GEN	libavdevice/libavdevice.pc
2024-05-20T00:10:38.7138156Z GEN	libavfilter/libavfilter.pc
2024-05-20T00:10:38.7277432Z CC	libavformat/3dostr.o
2024-05-20T00:10:38.8450485Z CC	libavformat/4xm.o
2024-05-20T00:10:38.8934157Z CC	libavformat/a64.o
2024-05-20T00:10:39.0070961Z CC	libavformat/aacdec.o
2024-05-20T00:10:39.0553998Z CC	libavformat/aadec.o
2024-05-20T00:10:39.1761982Z CC	libavformat/aaxdec.o
2024-05-20T00:10:39.3614222Z CC	libavformat/ac3dec.o
2024-05-20T00:10:39.4066402Z CC	libavformat/acedec.o
2024-05-20T00:10:39.5116541Z CC	libavformat/acm.o
2024-05-20T00:10:39.5420033Z CC	libavformat/act.o
2024-05-20T00:10:39.6314259Z CC	libavformat/adp.o
2024-05-20T00:10:39.6989635Z CC	libavformat/ads.o
2024-05-20T00:10:39.7658402Z CC	libavformat/adtsenc.o
2024-05-20T00:10:39.8236116Z CC	libavformat/adxdec.o
2024-05-20T00:10:39.9665111Z CC	libavformat/aea.o
2024-05-20T00:10:40.0697317Z src/libavcodec/x86/mathops.h: Assembler messages:
2024-05-20T00:10:40.0703056Z src/libavcodec/x86/mathops.h:125: Error: operand type mismatch for `shr'
2024-05-20T00:10:40.0704828Z src/libavcodec/x86/mathops.h:125: Error: operand type mismatch for `shr'
2024-05-20T00:10:40.0706452Z src/libavcodec/x86/mathops.h:125: Error: operand type mismatch for `shr'
2024-05-20T00:10:40.0708536Z src/libavcodec/x86/mathops.h:125: Error: operand type mismatch for `shr'
2024-05-20T00:10:40.0709965Z src/libavcodec/x86/mathops.h:125: Error: operand type mismatch for `shr'
2024-05-20T00:10:40.0711107Z src/libavcodec/x86/mathops.h:125: Error: operand type mismatch for `shr'
2024-05-20T00:10:40.0712618Z src/libavcodec/x86/mathops.h:125: Error: operand type mismatch for `shr'
2024-05-20T00:10:40.0713733Z src/libavcodec/x86/mathops.h:125: Error: operand type mismatch for `shr'
2024-05-20T00:10:40.0715010Z src/libavcodec/x86/mathops.h:125: Error: operand type mismatch for `shr'
2024-05-20T00:10:40.0716426Z src/libavcodec/x86/mathops.h:125: Error: operand type mismatch for `shr'
2024-05-20T00:10:40.0717626Z src/libavcodec/x86/mathops.h:125: Error: operand type mismatch for `shr'
2024-05-20T00:10:40.0718749Z src/libavcodec/x86/mathops.h:125: Error: operand type mismatch for `shr'
2024-05-20T00:10:40.0728614Z src/libavcodec/x86/mathops.h:125: Error: operand type mismatch for `shr'
2024-05-20T00:10:40.0730059Z src/libavcodec/x86/mathops.h:125: Error: operand type mismatch for `shr'
2024-05-20T00:10:40.0731435Z src/libavcodec/x86/mathops.h:125: Error: operand type mismatch for `shr'
2024-05-20T00:10:40.0733075Z src/libavcodec/x86/mathops.h:125: Error: operand type mismatch for `shr'
2024-05-20T00:10:40.0735925Z src/libavcodec/x86/mathops.h:125: Error: operand type mismatch for `shr'
2024-05-20T00:10:40.0754000Z src/libavcodec/x86/mathops.h:125: Error: operand type mismatch for `shr'
2024-05-20T00:10:40.0756460Z src/libavcodec/x86/mathops.h:125: Error: operand type mismatch for `shr'
2024-05-20T00:10:40.0843367Z CC	libavformat/afc.o
2024-05-20T00:10:40.0911289Z make: *** [/home/runner/.conan2/p/b/ffmpe346528a358a23/b/src/ffbuild/common.mak:67: libavformat/adtsenc.o] Error 1
2024-05-20T00:10:40.0914272Z make: *** Waiting for unfinished jobs....
2024-05-20T00:10:40.1620415Z 
2024-05-20T00:10:40.1653018Z ffmpeg/4.4.4: ERROR: 
2024-05-20T00:10:40.1658536Z Package 'b5b708f47d8ef22bc5a82002a804470f8d7787bc' build failed
2024-05-20T00:10:40.1659759Z ffmpeg/4.4.4: WARN: Build folder /home/runner/.conan2/p/b/ffmpe346528a358a23/b/build-release
2024-05-20T00:10:40.1660879Z ERROR: ffmpeg/4.4.4: Error in build() method, line 711
2024-05-20T00:10:40.1661484Z 	autotools.make()
2024-05-20T00:10:40.1662278Z 	ConanException: Error 2 while executing
2024-05-20T00:10:40.2457156Z ##[error]Process completed with exit code 1.
```

</details>

**Comment by uilianries** [comment]
@nagayev Hello! I'm not sure if your current error persists, but I can not reproduce it locally. 

> with disabled ssl.

Do you mean are you using the option `-o ffmpeg/*:with_ssl=False`? The same is not listed in your log and command usage in this issue. Also, please, share a full build log with your error, including the exactly command line used to build the package, every line counts, as there are many details in the log.

I can build ffmpeg/4.4.4 with `-o ffmpeg/*:with_ssl=False` without any error:

[ffmpeg-4.4.4-no-ssl.log](https://github.com/user-attachments/files/16244302/ffmpeg-4.4.4-no-ssl.log)

**Comment by GuigzLab** [comment]
I have the exact same problem here, unable to install ffmpeg/4.4.4 with your solution. I need it to build opencv/4.10.0

The only posts i found about this suggested to downgrade my version of binutils from 2.42 to 2.40 but I don't know how

**Comment by uilianries** [comment]
@GuigzLab Could you please post your entire build log? Also share the used OS, Conan version.

**Comment by uilianries** [comment]
Now I'm capable to reproduce the error, indeed is related to binutils 2.42 (Ubuntu 24.04). Here is my build log: 

[ffmpeg-4.4.4-shr-error.log](https://github.com/user-attachments/files/16579953/ffmpeg-4.4.4-shr-error.log)

The error is reported in the upstream as well: https://lists.ffmpeg.org/pipermail/ffmpeg-trac/2023-June/066203.html

And is fixed in the upstream too, and should be available in FFMpeg 7.1: https://github.com/FFmpeg/FFmpeg/commit/effadce6c756247ea8bae32dc13bb3e6f464f0eb

I'll checked to provide a backport patch to CCI recipe. Regards.

**Comment by uilianries** [comment]
Hello @nagayev ! I just addressed a fix to this issue on the PR #24905

/cc @GuigzLab 

Please, feel free to validate that hotfix!
